src/main/java/com/microsoft/graph/core/models/TokenValidable.java

```diff
@@ -0,0 +1,179 @@
+package com.microsoft.graph.core.models;
+
+import java.security.Key;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.Locator;
+
+/**
+ * TokenValidable interface
+ */
+public interface TokenValidable<U extends DecryptableContent, T extends EncryptedContentBearer<U>> {
+
+    /**
+     * Graph notification publisher. Ensures that a different app that isn't Microsoft Graph did not send the change notifications
+     */
+    public static final String graphNotificationPublisher = "0bf30f3b-4a52-48df-9a82-234910c4a086";
+
+    /**
+     * Sets collection of validation tokens
+     * @param validationTokens tokens
+     */
+    public void setValidationTokens(@Nullable final List<String> validationTokens);
+
+    /**
+     * Returns validation tokens
+     * @return list of tokens
+     */
+    public @Nullable List<String> getValidationTokens();
+
+    /**
+     * Sets collection of encrypted token bearers
+     * @param value collection of encrypted token bearers
+     */
+    public void setValue(@Nullable final List<T> value);
+
+    /**
+     * Get collection of encrypted token bearers
+     * @return encrypted token bearers
+     */
+    public @Nullable List<T> getValue();
+
+    /**
+     * Validates the tokens
+     * @param <U> DecryptableContent
+     * @param <T> EncryptedContentBearer
+     * @param collection collection of encrypted token bearers
+     * @param tenantIds tenant ids
+     * @param appIds app ids
+     * @param keyDiscoveryUrl the JWKS endpoint to use to retrieve signing keys
+     * @return true if the tokens are valid
+     * @throws IllegalArgumentException if one of the tokens are invalid
+     */
+    public static <U extends DecryptableContent, T extends EncryptedContentBearer<U>>
```

Comment:
I feel like those static methods could benefit from unit tests

Reply:
Updated these in the latest updates here.